### PR TITLE
feat: Search box debouncing 

### LINKED ISSUE #685: [Feat]:  search box performance improve
### What feature?

I have an idea 
we can use debouncing in search box to improve performance 

what is debouncing in search box - 
In the context of a search box, debouncing is a technique used to control and optimize the execution of search queries or other related actions triggered by user input. When a user types into a search box, it often results in frequent events firing, such as keypress or input events. Each event may trigger an action, such as sending a request to a server to retrieve search results or updating the search suggestions

### Add screenshots

Add screenshots

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mandliyarajendra11**
## Related Issue

Closes #685 

## Description

what is deboncing 
Debouncing in JavaScript is a practice used to improve browser performance. There might be some functionality in a web page that requires time-consuming computations. If such a method is invoked frequently, it might greatly affect the performance of the browser, as JavaScript is a single-threaded language. 

**check this video for better understanding -** 
https://youtu.be/qHKHASOuuFA

## Checklist

- [ ] My code adheres to the established style guidelines of the project.
- [ ] I have included comments in areas that may be difficult to understand.
- [ ] My changes have not introduced any new warnings.
- [ ] I have conducted a self-review of my code.

### CODE REVIEW

File: src/components/Navbar.jsx [hunk 1]
```diff
@@ -45,7 +47,18 @@ const Header = ({ filteredSuggestions }) => {
   }
   const handleChangeInInput = (event) => {
     const val = event.target.value
-    setSearchTerm(val)
+    setSearch(val)
+    
+
+  const debounce = (fn, delay) => {
```
Comment (VenomFate-619) [inline src/components/Navbar.jsx:53]: move this function to the utils file.

**Comment by JasonDsouza212** [comment]
@mandliyarajendra11 the PR has unrelated changes

**Comment by mandliyarajendra11** [comment]
> @mandliyarajendra11 the PR has unrelated changes

mean i have to move this fun in utils file and  need to solve conflict

**Comment by JasonDsouza212** [comment]
@mandliyarajendra11 your PR is deleting and making changes in the product.json file which is not part of the issue

**Pull request closed without merging**